pkg/virt-handler/vm.go

```diff
@@ -2689,3 +2688,33 @@ func setMissingSRIOVInterfacesNames(interfacesSpecByName map[string]v1.Interface
 		}
 	}
 }
+
+func (d *VirtualMachineController) isHostModelMigratable(vmi *v1.VirtualMachineInstance) error {
+	if cpu := vmi.Spec.Domain.CPU; cpu != nil && cpu.Model == v1.CPUModeHostModel {
+		if !d.clusterConfig.HostModelCPUEnabled() {
+			return fmt.Errorf("vmi \"%s\" uses host-model CPU mode while feature gate \"%s\" is disabled",
+				vmi.Name, virtconfig.HostModelCPUMigratableFeatureGate)
+		}
+
+		node, err := d.clientset.CoreV1().Nodes().Get(context.Background(), vmi.Status.NodeName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if !nodeHasHostModelLabel(node) {
+			return fmt.Errorf("vmi \"%s\" uses host-model CPU mode and tries to migrate into node \"%s\" "+
```

Comment:
I think this error message is a bit confusing. The issue is that we cannot translate what host-model means on this node and therefore we cannot migrate this VMI.
Maybe something like `The node %s has no (%s/...) label to allow migration with host-model.
You could also add a log with a longer message perhaps?